text/0118-search-api.md

```diff
@@ -108,156 +109,355 @@ It returns the number of documents matching the current search query for each gi
+
+If `"*"` is provided as a value: `attributesToCrop=["*"]` all the attributes present in `attributesToRetrieve` will be automatically assigned to `attributesToCrop`.
 
-When this parameter is set, returned documents include a `_formatted` object containing the cropped terms.
+- 🔴 Sending a value with a different type than `Array[String]`(POST), `String`(GET) or `null` for `attributesToCrop` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
 
-Optionally, indicating a custom crop length for any of the listed attributes is possible: `attributesToCrop=["attributeNameA:25", "attributeNameB:150"]`. The custom crop length set in this way has priority over the `cropLength` parameter.
+##### 3.1.11.1. Custom `cropLength` Defined Per Attribute.
 
-Instead of supplying individual attributes, it is possible to provide `["*"]` as a value: `attributesToCrop=["*"]`. This will crop the values of all attributes present in `attributesToRetrieve`.
+Optionally, indicating a custom crop length for any of the listed attributes is possible:
 
-- 🔴 Sending a value with a different type than `Array[String]`(POST), `String`(GET) or `null` for `attributesToCrop` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+`"attributesToCrop":["attributeNameA:15", "attributeNameB:30"]`
 
-> See [_Formatted Field Behavior](0039-_formatted-field-behavior_.md)
+A custom crop length set in this way has priority over the `cropLength` parameter.
 
-##### 1.2.1.10 `cropLength`
+##### 3.1.11.2. searchableAttributes
+
+As for `attributesToHighlight`, attributes not defined in the `searchableAttributes` index setting are also cropped if assigned to `attributesToCrop`.
+
+##### 3.1.11.3. stopWords
+
+Attributes defined in the `stopWords` index setting are counted as words regarding `cropLength`.
+
+##### 3.1.11.3. Tokenizer Separators
+
+Tokenizer separators aren't counted as words regarding `cropLength`.
+
+#### 3.1.12. `cropLength`
 
 - Type: Integer
 - Required: False
-- Default: `200`
+- Default: `10`
+
+Sets the total number of **words** to keep for the cropped part of an attribute specified in the `attributesToCrop` parameter. It means that if `10` is set for `cropLength`, the cropped part returned in `_formatted` will only be 10 words long.
+
+This parameter is applied to the fields from `attributesToCrop`. If there are none, this parameter has no effect. See [3.1.11. `attributesToCrop`](#3111-attributestocrop) section.
+
+- 🔴 Sending a value with a different type than `Integer` or `null` for `cropLength` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+##### 3.1.12.1 Examples
+
+###### 3.1.12.1.1. Extending around
+
+Given an attribute defined in `attributesToCrop` containing:
+
+`"In his ravenous hatred he found no peace, and with boiling blood he scoured the umbral plains, seeking vengence afgainst the dark lords who had robbed him."`
```

Comment:
What happens if I search for `he found no peace` but I set `cropLength` to 2 ? 
And what if I set it to 0 ? Is it considered as null ?